docs/rfcs/0004-suggested-excluded-apps-tech-spec.md

```diff
@@ -0,0 +1,141 @@
+-   How hard would these checks be to run in Android daemon or PeriodicWorkRequest? (If substantially harder, Santiago open to doing it in Android app - which reduces how often it runs, as the app must be open in order for it to run.) On desktop platforms, the app is always active when the VPN is active. This is not true on mobile, which is why this is considered separately.
+-   Any easy ways to categorize games? (See [VPN-4433](https://mozilla-hub.atlassian.net/browse/VPN-4433).)
+-   How will we decide which apps are initially on the suggested list?
+-   Does the suggested list just include specific apps, or can it include entire directories (like well-known Windows game directories) and/or allow wildcard matching?
+
+Implementation (estimated 39-64 points total)
+---------------------------------------------------
+
+This work is broken into four sections, with the bulk of code coming in the middle two sections. Those two sections should each be built behind their own separate feature flag, though we'd expose both parts together via Nimbus.
+
+### Prerequisites (21-36 points total)
+
+-   (3-5 points) Answer the open questions above
+-   (5-8 points) Create a well-structured file (for our own research, as well as one users can add to via GH pull requests on the client repo) w/ apps to exclude (and update the repo's readme to explain this new type of contribution). Add this doc to addons. Add tests to ensure the doc's format integrity is maintained. This format should be designed to allow new platforms (like websites) to be added without breaking anything.
+    -   Current proposal for format: Individual text files for each platform (`android.txt`, `linux.txt`, etc.) in a specific directory. Each line of the file will be a single path, or bundle ID, or whatever we end up using as an identifier. (See "Other options considered" section for why this is the current proposal.)
+    -   Additions to the list should be given the same amount of thought that adding new dependencies to our repo are given. That is, full research by our engineers to ensure nothing malicious is being snuck in. And erring on the side of not including any.
+        -   This statement (or something similar) should be added to the top of the file in a comment.
+    -   From Santiago: <https://github.com/citizenlab/test-lists/tree/master/lists>, <https://ooni.org/>
+    -   Santiago list for Android: <https://docs.google.com/spreadsheets/d/1j2REPQkcdPg5bNW5B3T-L8dnlNPPwAms1WlBiYvVwoc/edit#gid=0>
+-   (8-13 points) Deep linking into app for notification to work ([VPN-5034](https://mozilla-hub.atlassian.net/browse/VPN-5034)): Currently, our navigation code allows us to switch tabs, but does not permit moving to a deeper screen - so we can move a user to the settings screen, but can't move them into a menu or submenu within the screen (like App Exclusions). We need deep linking to be able to open the notification where we want.
```

Comment:
We can refactor the App Exclusions view to be a `Screen` and call it easily enough with the Navigator as we do for `ScreenGetHelp` and others.

Reply:
That only allows us to present the screen, right?

If so - We may want to be switching to the Settings tab beforehand (awaiting word from Santiago). That's possible, but not in the app today, right?

Reply:
> That only allows us to present the screen, right?

Correct. A truly nested deep link (in this case, where clicking back takes you to the settings main menu) is something that would have to be implemented.

Reply:
Confirming w/ Santiago how he wants to handle this.